dd-sdk-android/src/test/kotlin/com/datadog/android/error/internal/CrashReportsFeatureTest.kt

```diff
@@ -350,4 +352,31 @@ internal class CrashReportsFeatureTest {
         assertThat(CrashReportsFeature.dataUploadScheduler)
             .isInstanceOf(NoOpUploadScheduler::class.java)
     }
+
+    @Test
+    fun `clears all files on local storage on request`(
+        @StringForgery(StringForgeryType.NUMERICAL) fileName: String,
+        @StringForgery(StringForgeryType.ALPHABETICAL) content: String
+    ) {
+        val fakeDir = File(rootDir, CrashLogFileStrategy.CRASH_REPORTS_FOLDER)
+        fakeDir.mkdirs()
+        val fakeFile = File(fakeDir, fileName)
+        fakeFile.writeText(content)
+
+        // when
+        CrashReportsFeature.initialize(
+            mockAppContext,
+            fakeConfig,
+            mockOkHttpClient,
+            mockNetworkInfoProvider,
+            mockUserInfoProvider,
+            mockTimeProvider,
+            mockSystemInfoProvider,
+            mockScheduledThreadPoolExecutor
+        )
+        CrashReportsFeature.clearAllData()
+
+        // Then
+        assertThat(fakeFile).doesNotExist()
```

Comment:
How does the assertion wait for the `FileReader's` thread to finish deletions 🤔?

Reply:
Is it synchronous? If so, shouldn't we dispatch it from the reader's thread to avoid race conditions and unexpected behaviours?

Reply:
That call is synchronous, but indeed dispatching it to the reader's thread might be a good idea